### PR TITLE
PD-665: Upgrade to react-transition-group 4.4.x which fixes StrictMode warnings

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Upgrade to react-transition-group 4.4.x which fixes StrictMode warnings
- Validates that animations still work
- Validates that strict mode console warnings no longer appear
🎟 _Jira ticket:_ [PD-665](https://jira.mongodb.org/browse/PD-665)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: .changeset/cyan-plums-grow.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@leafygreen-ui/menu': patch
+'@leafygreen-ui/modal': patch
+'@leafygreen-ui/popover': patch
+---
+
+Upgrades react-transition-group to 4.4.1
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply .changeset/cyan-plums-grow.md:7]: Sounds good to me

File: packages/menu/src/SubMenu.tsx [hunk 2]
```diff
@@ -222,7 +222,7 @@ const SubMenu = React.forwardRef((props: SubMenuProps, ref) => {
   } = props;
 
   const { usingKeyboard: showFocus } = useUsingKeyboardContext();
-
+  const nodeRef = React.useRef(null);
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/menu/src/SubMenu.tsx:225]: In SubMenu?

[the remark this answers is not in the document]
Comment (hswolff) [reply packages/menu/src/SubMenu.tsx:225]: Yeah

[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/menu/src/SubMenu.tsx:225]: I can't repro locally

File: packages/modal/src/Modal.tsx [hunk 3]
```diff
@@ -194,10 +195,17 @@ function Modal({
   useEscapeKey(handleClose, { enabled: open });
 
   return (
-    <Transition in={open} timeout={500} mountOnEnter unmountOnExit>
+    <Transition
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/modal/src/Modal.tsx:198]: I don't know why this timeout was ever set to 500, as all of the transitions within the file take 150ms... 🤔 I've fixed

**Pull request merged**